capten/agent/pkg/capten-store/argocd_projects.go

```diff
@@ -102,8 +102,24 @@ func (a *Store) updateArgoCDProjects() ([]*model.ArgoCDProject, error) {
 	}
 
 	argoCDProjects := make(map[string]*model.ArgoCDProject)
-	for _, tekPro := range regArgoCDProjects {
-		argoCDProjects[tekPro.Id] = tekPro
+	for _, argoCDPro := range regArgoCDProjects {
+		var deleteArgoCDRecord bool
+		for _, gitProject := range gitProjects {
+			if gitProject.Id == argoCDPro.GitProjectId {
+				deleteArgoCDRecord = false
+				break
+			} else {
+				deleteArgoCDRecord = true
+			}
+		}
+
+		if deleteArgoCDRecord {
+			if err := a.DeleteArgoCDProjectsData(argoCDPro.Id); err != nil {
+				return nil, err
+			}
+		} else {
+			argoCDProjects[argoCDPro.GitProjectId] = argoCDPro
+		}
 	}
 
 	ret := make([]*model.ArgoCDProject, 0)
```

Comment:
check first if exist in map, if exist use ap project from the map,
if not exist, create and add the project.

do the same for other methosds

project := &model.ArgoCDProject{Id: gitProject.Id, GitProjectId: gitProject.Id,
			GitProjectUrl: gitProject.ProjectUrl}
		if ap, ok := argoCDProjects[gitProject.Id]; !ok {
			project.Status = string(model.ArgoCDProjectAvailable)
			project.LastUpdateTime = time.Now().Format(time.RFC3339)
			if err := a.UpsertArgoCDProject(project); err != nil {
				return nil, err
			}
		} else {
			project.Status = argoCDProjects[gitProject.Id].Status
			project.LastUpdateTime = ap.LastUpdateTime
		}